gateway/server.go

```diff
@@ -877,7 +875,9 @@ func (gw *Gateway) isRPCMode() bool {
 
 func (gw *Gateway) rpcReloadLoop(rpcKey string) {
 	for {
-		gw.RPCListener.CheckForReload(rpcKey)
+		if ok := gw.RPCListener.CheckForReload(rpcKey); !ok {
+			return
+		}
```

Comment:
CheckForReload: changed return to bool signature to respect cancellation context of gateway